flink-table-store-core/src/test/java/org/apache/flink/table/store/file/mergetree/compact/CompactManagerTest.java

```diff
@@ -197,7 +197,7 @@ private void innerTest(
                 new CompactManager(
                         service, strategy, comparator, 2, testRewriter(expectedDropDelete));
         manager.submitCompaction(levels);
-        manager.finishCompaction(levels);
```

Comment:
Add tests so that compaction thread will randomly hang to simulate heavy compaction tasks.